packages/roosterjs-editor-core/lib/coreApi/ensureUniqueId.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Add an unique id to element and ensure that is unique
+ * @param el The HTMLElement that will receive the id
+ * @param idPrefix The prefix that will antecede the id (Ex: tableSelected01)
+ */
+export function ensureUniqueId(el: HTMLElement, idPrefix: string) {
+    const doc = el.ownerDocument;
+
+    if (!el.id) {
+        let cont = 0;
+        const getElement = () => doc.getElementById(idPrefix + cont);
+        //Ensure that there are no elements with the same ID
+        let element = getElement();
+        while (element) {
+            cont++;
+            element = getElement();
+        }
+
+        el.id = idPrefix + cont;
+    } else {
+        const elements = doc.querySelectorAll(`#${el.id}`);
```

Comment:
I know this is moved from table selection, but let's refine it. It may not be a good idea to call the function recursively. So we can do:

```
if (has id) {
  if (id is not unique) {
     remove id;
  }
}
if (not have id) {
   apply an id
}
```

Reply:
I fixed and changed the name to addUniqueId, since the function is responsible to add the id and not only ensure that is unique.